datafusion/sql/src/select.rs

```diff
@@ -241,33 +236,26 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
         plan: LogicalPlan,
         outer_query_schema: Option<&DFSchema>,
         planner_context: &mut PlannerContext,
-        from_schema: &DFSchema,
     ) -> Result<LogicalPlan> {
         match selection {
             Some(predicate_expr) => {
                 let mut join_schema = (**plan.schema()).clone();
-                let mut all_schemas: Vec<DFSchemaRef> = vec![];
-                for schema in plan.all_schemas() {
-                    all_schemas.push(schema.clone());
-                }
-                if let Some(outer) = outer_query_schema {
-                    all_schemas.push(Arc::new(outer.clone()));
+
+                let fallback_schemas = plan.fallback_normalize_schemas();
+                let outer_query_schema = if let Some(outer) = outer_query_schema {
                     join_schema.merge(outer);
-                }
-                let x: Vec<&DFSchemaRef> = all_schemas.iter().collect();
+                    vec![outer]
+                } else {
+                    vec![]
+                };
 
```

Comment:
The join schema logic is problematic. I will fix it in other PR.  The filter_expr might contains